File: C/2028.c
#include <stdio.h>
#include <math.h>

int digit(int n)
{
	int x;
	x = n;
	while (x > 0){
		x /= 10;
		x += 1;
	}
	return x;
}

int construct(int num, int digits){
	int flag = 0, ans = 0;
	while (flag < digits){
		ans += (num % 10) * (pow(10, flag));
		flag += 1;
	}
	return ans;
}

int main()
{
	int tc, i, k, a, square;
	scanf("%d", &tc);
	for (i = 1; i <= tc; i++){
		scanf("%d", &k);
		square = k * k;
		if ((construct(square, digit(k))) == k){
			printf("YES");
		}
		else
			printf("NO");
	}

	return 0;
}

File: Python/2028.py
class Square:
	def __init__(self, N):
		self._n = N
		self._digits = 0

	def digit(self):
		x = self._n
		while x > 0:
			x //= 10
			x += 1
		self._digits = x

	def construct(self, num, digits):
		flag = 0
		ans = 0
		while flag < digits:
			ans += (num % 10) * (10 ** flag)
			flag += 1
		return ans

	def compare(self):
		self.digit()
		square = self._n ** 2
		if self.construct(square, self._digits) == self._n:
			print("YES")
		else:
			print("NO")

tc = int(input())
for i in range(tc):
	k = int(input())
	Square(k).compare()





